Swiftfin/Views/ItemEditorView/Components/RefreshMetadataButton.swift

```diff
@@ -0,0 +1,175 @@
+//
+// Swiftfin is subject to the terms of the Mozilla Public
+// License, v2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at https://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) 2024 Jellyfin & Jellyfin Contributors
+//
+
+import JellyfinAPI
+import SwiftUI
+
+extension ItemEditorView {
+
+    struct RefreshMetadataButton: View {
+
+        @StateObject
+        private var viewModel: RefreshMetadataViewModel
+
+        @State
+        private var isPresentingEventAlert = false
+        @State
+        private var error: JellyfinAPIError?
+        @State
+        private var isFinishing = false
```

Comment:
There isn't a way to query the task that's spun up for refreshing the metadata. It's good to see some attempt at UX for it but I don't think we could provide a good enough abstraction for refreshing the metadata. I would consider this an uncommon enough task and that admins are somewhat understanding that the item must be popped and reselected in order to see the new metadata.

Reply:
Sounds good. Are we fine with this just disabling the button for 2 seconds and showing the checkmark? Just some sort of feedback to prevent this being spammed on accident thinking nothing happened.

Reply:
Here's the version I have now that waits 5 seconds before pulling the new item:

https://github.com/user-attachments/assets/7e980ea3-7d84-4611-bf3d-9a65b8cba70b